docs/scos/dev/front-end-development/202307.0/oryx/building-components/oryx-implementing-components.md

```diff
@@ -0,0 +1,219 @@
+
+Component options are resolved by `ContentMixin`, similar to how `ProductService` resolves the product data. You can combine multiple mixins in a component implementation—for example:
+
+```ts
+import { resolve } from "@spryker-oryx/di";
+import { ContentMixin } from "@spryker-oryx/experience";
+import { ProductMixin } from "@spryker-oryx/product";
+
+interface ProductIdOptions {
+  myOption?: boolean;
+}
+
+export class ProductIdComponent extends ProductMixin(
+  ContentMixin<ProductIdOptions>(LitElement)
+) {
+  protected override render(): TemplateResult {
+    const { myOption } = this.$options();
+
+    if (!myOption) return;
+
+    return html`id: ${this.$product()?.id}`;
+  }
+}
+```
+
+You can provide default options in the component, in feature sets, or in the application. For more details, see [Component options](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-components/component-options.html).
+
+## 4. Styling the component DOM
+
+Oryx components are styled with standard CSS. The components have a separate DOM attached using the open [shadow DOM](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM). The shadow DOM encapsulates the styles so that they cannot _leak_ into other components and prevents global styles from cascading down to the component.
+
+Styling components in the shadow DOM is a big topic we recommend studying separately. However, there are a few things to know when it comes to Oryx and styling components:
+
+- Design system components are provided in the UI package. Components like `<oryx-button>` or `<oryx-link>` are used to ensure a common visual language. They can be customized.
+- Font styles rules like `font-face` or `font-size`, unlike other CSS rules, cascade to web components, no matter how deep they are nested. Standard font rules are provided in the `<oryx-app>` component.
+- [Typography](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-applications/styling/oryx-typography.html) design tokens are used to ensure consistent styling.
+- Custom properties, also known as CSS variables, cascade into web components, which is why the application theme is based on CSS variables. See [Design tokens](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-applications/styling/oryx-design-tokens.html) for more information.
+- Oryx provides an [icon system](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-applications/styling/oryx-icon-system.html) that you can leverage in your components.
```

Comment:
Rephrase this to make it relevant to styling the component

Reply:
rephrased a bit, but I was quit ok with the text already. We might have a disconnect here.